public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,885 @@
+
+:marked
+  Call it in the constructor, after creating the form.
++makeExample('reactive-forms/ts/app/hero-detail.component.ts', 'ctor')(format=".")
+
+:marked
+  The `watchName` method pushes name-change values into a `nameChangeLog` array.
+  Display that array at the bottom of the component template with this `*ngFor` binding:
++makeExample('reactive-forms/ts/app/hero-detail.component.html', 'name-change-log','app/hero-detail.component.html (Name change log)')(format=".")
+
+:marked
+  Return to the browser, select a hero (e.g, "Magneta"), and start typing in the _name_ input box.
+  You should see a new name in the log after each keystroke.
+
+  ### When to use it
+  
+  An interpolation binding is the easier way to _display_ a name change.
+  Subscribing to an observable form control property is handy for triggering 
+  application logic _within_ the component class.
+
+.l-main-section
+a#save
+:marked
+  ## Save form data
+
+  The `HeroDetailComponent` captures user input but it doesn't do anything with it.
+  In a real app, you'd probably save those hero changes.
+  In a real app, you'd also be able to revert unsaved changes and resume editing.
+  After you implement both features in this section, the form will look like this:
+
+figure.image-display
+  img(src="/resources/images/devguide/reactive-forms/save-revert-buttons.png" width="389px" alt="Form with save & revert buttons")
+
+:marked
+  ### Save
+  In this sample application, the `HeroDetailComponent` cannot perform the save operation. 
+  The component delegates that chore to its parent `HeroListComponent`.
+  
+  When the user clicks the _Save_ button, the `HeroDetailComponent` raises a save event on its `save` output property.
+  The parent `HeroListComponent`, which binds to that property, responds to the event and processes the event payload.
++makeExample('reactive-forms/ts/app/hero-list.component.html', 'hero-binding','app/hero-list.component.html (save event binding)')(format=".")
+:marked
+  The save event payload _could_ have any shape but often it's in the shape of the data model and that is the case in this example.
++makeExample('reactive-forms/ts/app/hero-detail.component.ts', 'save-emitter')(format=".")
+
+:marked
+  The `HeroDetailComponent` save method (`onSubmit`) constructs a `saveHero`
+  and emits a save message, with `saveHero` as the payload, for the `HeroListComponent` to process.
+
++makeExample('reactive-forms/ts/app/hero-detail.component.ts', 'on-submit','app/hero-detail.component.ts (onSubmit)')(format=".")
+:marked
+  In the reactive paradigm, the _data model_ is immutable so the save method can't update the component's `hero`
+  and send that as the save event payload.
+
+  Instead, it must prepare `saveHero` whose values derive from a combination of original hero values (the `hero.id`)
+  and deep copies of the potentially-changed form model values.
```

Comment:
The reactive form directives aren't prescriptive. I've been told, however, that you felt strongly about avoiding data model mutation when using the reactive style.

We can crisp this up so it is less pedantic. Let me take a look at that.